[12.x]Fix PHPDoc return types

mohammadrasoulasghari opened the pull request:
In this PR, I've made three improvements to enhance code clarity and type accuracy without changing any functionality.🧹✨

### 1. Updated `forceDestroy()` return type  to `int`

The `forceDestroy()` method directly returns the result of `destroy()`, which returns an integer (the count of deleted records). Updated the PHPDoc to accurately reflect this relationship for better type consistency.

### 2. Aligned `resolveConnection()` return type with interface definition

Updated the return type from `Connection` to `ConnectionInterface` to match the actual return type from the underlying resolver's `connection()` method. This ensures type consistency and improves IDE support when working with connection objects.


I noticed these issues while refactoring the Model class and thought it would be beneficial to update them for better code quality and developer experience.🔍👨‍💻

mohammadrasoulasghari committed:
fix: correct return type of forceDestroy method to int

The forceDestroy method directly returns the result of the destroy method,

rodrigopedra commented:
> 3. Simplified conditional logic using logical OR operator
> 
> Replaced a ternary conditional expression with a more readable logical OR operator that maintains identical behavior while making the code more concise and **easier to understand**.

I guess the "easier to understand" argument is very subjective here.

I find the previous code easier to understand, for these reasons:

- It is more declarative, as it doesn't rely on knowledge of how a boolean operator short-circuits
- It doesn't rely on an inverted logic by using the `!` (not) operator
- Thus, for newer devs, it doesn't raise questions on why the condition couldn't be flipped

I am all in for the PHP docs fixes, but I feel this other change should be sent as a separate PR.

mohammadrasoulasghari committed:
fix: align resolveConnection return type with ConnectionResolverInterface

Updated the PHPDoc return type from Connection to ConnectionInterface
to match the actual return type from the underlying resolver's connection method.
This provides better type consistency across the codebase.

mohammadrasoulasghari commented:
> > 3. Simplified conditional logic using logical OR operator
> > 
> > Replaced a ternary conditional expression with a more readable logical OR operator that maintains identical behavior while making the code more concise and **easier to understand**.
> 
> I guess the "easier to understand" argument is very subjective here.
> 
> I find the previous code easier to understand, for these reasons:
> 
> * It is more declarative, as it doesn't rely on knowledge of how a boolean operator short-circuits
> * It doesn't rely on an inverted logic by using the `!` (not) operator
> * Thus, for newer devs, it doesn't raise questions on why the condition couldn't be flipped
> 
> I am all in for the PHP docs fixes, but I feel this other change should be sent as a separate PR.

Thank you for your feedback!🙏 

You’re right, I’ve reverted the change back to the original ternary expression.

mohammadrasoulasghari commented:
> > 3. Simplified conditional logic using logical OR operator
> > 
> > Replaced a ternary conditional expression with a more readable logical OR operator that maintains identical behavior while making the code more concise and **easier to understand**.
> 
> I guess the "easier to understand" argument is very subjective here.
> 
> I find the previous code easier to understand, for these reasons:
> 
> * It is more declarative, as it doesn't rely on knowledge of how a boolean operator short-circuits
> * It doesn't rely on an inverted logic by using the `!` (not) operator
> * Thus, for newer devs, it doesn't raise questions on why the condition couldn't be flipped
> 
> I am all in for the PHP docs fixes, but I feel this other change should be sent as a separate PR.

You’re right, this is indeed subjective. I’ve reverted the change back to the original ternary expression.🙏

taylorotwell commented:
It's fine.
